connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/lifecycle/InboundConnectorRestController.java

```diff
@@ -47,34 +45,44 @@ public List<ActiveInboundConnectorResponse> getActiveInboundConnectors(
       @RequestParam(required = false) String type) {
     var result =
         inboundManager.query(new ActiveInboundConnectorQuery(bpmnProcessId, elementId, type));
-    return result.stream().map(this::mapToResponse).collect(Collectors.toList());
+    return result.stream().map(this::mapToInboundResponse).collect(Collectors.toList());
   }
 
-  private ActiveInboundConnectorResponse mapToResponse(ActiveInboundConnector connector) {
+  @GetMapping("/inbound/{bpmnProcessId}/{elementId}/{tenantId}/logs")
+  public List<Queue<ActivityLog>> getActiveInboundConnectorLogs(
+      @PathVariable String bpmnProcessId,
+      @PathVariable String elementId,
+      @PathVariable String tenantId) {
+    var result =
+        inboundManager.query(new ActiveInboundConnectorQuery(bpmnProcessId, elementId, null));
+    return result.stream()
+        .filter(r -> tenantIdMatch.test(r, tenantId))
+        .map(this::mapToInboundLogsResponse)
+        .collect(Collectors.toList());
+  }
+
+  BiPredicate<ActiveInboundConnector, String> tenantIdMatch =
+      (connector, tenantId) -> {
+        var definition = connector.context().getDefinition();
+        return tenantId != null && tenantId.equals(definition.tenantId());
+      };
+
+  private Queue<ActivityLog> mapToInboundLogsResponse(ActiveInboundConnector connector) {
+    var logs = connector.context().getLogs();
+    return logs;
+  }
```

Comment:
Not sure whether this requires a dedicated method

Reply:
You are right, will remove it.